src/Roslyn.Diagnostics.Analyzers/Core/SymbolIsBannedAnalyzer.cs

```diff
@@ -73,20 +75,55 @@ private static void OnCompilationStart(CompilationStartAnalysisContext compilati
                 return;
             }
 
-            var bannedApis = ResolveBannedApis(compilationContext.Compilation, apiData, compilationContext.CancellationToken);
-            if (bannedApis.Count > 0)
+            var bannedSymbols = ResolveBannedApis(compilationContext.Compilation, apiData, compilationContext.CancellationToken);
+
+            if (bannedSymbols.Count > 0)
             {
                 var symbolDisplayFormat = compilationContext.Compilation.Language == LanguageNames.CSharp
                     ? SymbolDisplayFormat.CSharpShortErrorMessageFormat
                     : SymbolDisplayFormat.VisualBasicShortErrorMessageFormat;
+
+                var bannedAttributes = bannedSymbols
+                    .Where(s => s is ITypeSymbol n && n.IsAttribute())
+                    .ToImmutableHashSet();
+
+                if (bannedAttributes.Count > 0)
+                {
+                    compilationContext.RegisterCompilationEndAction(context =>
+                    {
+                        VerifyAttributes(context.ReportDiagnostic, compilationContext.Compilation.Assembly.GetAttributes());
+                        VerifyAttributes(context.ReportDiagnostic, compilationContext.Compilation.SourceModule.GetAttributes());
+                    });
+
+                    compilationContext.RegisterSymbolAction(
+                        sac => VerifyAttributes(sac.ReportDiagnostic, sac.Symbol.GetAttributes()),
+                        SymbolKind.NamedType,
+                        SymbolKind.Method,
+                        SymbolKind.Field,
+                        SymbolKind.Property,
+                        SymbolKind.Event);
+                }
+
                 compilationContext.RegisterOperationAction(
-                    oac => AnalyzeOperation(oac, bannedApis, symbolDisplayFormat),
+                    oac => AnalyzeOperation(oac, bannedSymbols, symbolDisplayFormat),
                     OperationKind.ObjectCreation,
                     OperationKind.Invocation,
                     OperationKind.EventReference,
                     OperationKind.FieldReference,
                     OperationKind.MethodReference,
                     OperationKind.PropertyReference);
+
+                void VerifyAttributes(Action<Diagnostic> reportDiagnostic, ImmutableArray<AttributeData> attributes)
```

Comment:
Consider - allowing the banning of specific attribute constructors, but not others.

Reply:
For sure. I'd like to allow banning any overloadable member correctly. I think that's likely to be a fairly common requirement.

Reply:
Support exists for banning specific fields/properties/constructors/methods/events.